api/routes/search.py

```
@@ -0,0 +1,52 @@
+
+@router.get("/", response_model=SearchResults, status_code=200)
+async def search_items(query: str, request: Request):
+    """Elasticsearch에서 아이템 검색"""
+    cache_key = f"search:{query}"
+    redis_client = getattr(request.app.state, "redis", None)
+
+    # Attempt to fetch from cache first
+    if redis_client is not None:
+        try:
+            cached = await redis_client.get(cache_key)
+        except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)